### ISSUE TITLE
SlidingScreen is likely to be replaced with TransitionNode

### ISSUE THREAD

**Issue by samreid** [phetsims/equality-explorer#75]
From https://github.com/phetsims/twixt/issues/10

> SlidingScreen was ported from Make a Ten (though still exists there as a copy), and is used by Equality Explorer. TransitionNode is more general and flexible, and leverages Animation.js which is our proposed foundation for this kind of behavior. We should delete SlidingScreen and replace usages with TransitionNode.

Transition and TransitionNode are still in review as part of https://github.com/phetsims/twixt/issues/11 .  Once review is complete, it is very likely that we will deprecate SlidingScreen and use TransitionNode instead.  Just giving a heads up since I noticed that Equality Explorer uses SlidingScreen in SolveItScreenView.  I'll put this issue on hold until the review is further along.

**Comment by pixelzoom** [comment]
+1 to deprecate SlidingScreen and use TransitionNode.  Preferably well before Equality Explorer enters testing phase.

**Comment by pixelzoom** [comment]
Addressing this issue is not a prerequisite to publishing 1.0.0 (but sure would be nice).

**Comment by pixelzoom** [comment]
@jonathanolson In https://github.com/phetsims/make-a-ten/issues/283#issuecomment-389718351 I see that you replaced Make A Ten's `SlidingScreen` (it's own copy) with TWIXT/TransitionNode.   If I follow same pattern for converting, will it work here, where I'm using TWIXT/SlidingScreen?

**Comment by jonathanolson** [comment]
> If I follow same pattern for converting, will it work here, where I'm using TWIXT/SlidingScreen?

I'd expect so, the API for both SlidingScreens was quite similar. I'd also be interested in any feedback on TransitionNode.

**Comment by pixelzoom** [comment]
In the above commit, I converted from SlidingScreen to TransitionNode, following what was done in MakeATenGameScreenView.   The "slide" from level-selection to challenges is not working, it's an abrupt jump.  @jonathanolson what am I doing wrong?

**Comment by pixelzoom** [comment]
Hmmm... I guess there is a bit of a "slide".  But there's a pause first, followed by an abrupt slide.  Not the same smooth slide from challenges to level-selection.   I'm using identical options for both slide directions.

**Comment by pixelzoom** [comment]
This was (is) a performance problem.  

TransitionNode uses addChild and removeChild to change its content.  In the case of Equlity Explorer, `scenesParent` contains Nodes for all scenes, not just the selected scene, and visibility is used to show the selected scene. But  when this is executed:

`self.transitionNode.slideLeftTo( selectedSceneNode, TRANSITION_OPTIONS );`

... TransitionNode does `addChild( selectedSceneNode )`.  That takes a relatively long time.  So there's a pause, followed by a `step` with a big dt, which make the slide look jerky.

This is a general problem that has been moved to https://github.com/phetsims/twixt/issues/17. 

For EqEx, I've rewritten things so that only the Node for the selected scene is added to the scenegraph.  This yields acceptable performance. 

Closing.
